### DISCUSSION TITLE
there's any planning for reducing blazor wasm publish size?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/runtime#95239] [Q&A]
hi everyone,
im an iranian people i dont know anything about download speed in your countries
but in my region network speed usualy between 1-2.5mb in second
blazor wasm applications with brotli have 6-9mb size
this means our users must be wait at least 10 second to downloding whole application then wait 2-5 second for loading .net runtime in browser.
theres no any plan for this problem?
to reducing blazor wasm full application size to 1-2mb?
i think we need to do something for this problem like NAot for compiling and running blazor without .net runtime in browser environment
i wana to help .net community but i dont how to find problems and which tasks is available to do.

**Comment by Clockwork-Muse** [comment]
As was mentioned in [the original discussion you opened](https://github.com/dotnet/runtime/discussions/94636), work on compiling to JS modules is ongoing.

You can follow some of the [guidance for this area](https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-8.0) to see if it helps your usecase - You can enable NativeAOT (at a potential size penalty), or deploy a trimmed version of `dotnet.wasm` yourself (with the downside that it won't be a shared framework that other applications could download and use).

If you don't actually need everything to execute on the client side, you could move to a server-side only model, which would decrease the data required on first load (this may not be a savings over the long run, however).
